config/value.go

```diff
@@ -229,41 +237,41 @@ func (cv Value) TryAsFloat() (float64, bool) {
 // AsString returns the configuration value as a string, or panics if not
 // string-able
 func (cv Value) AsString() string {
-	s, ok := cv.TryAsString()
-	if !ok {
-		panic(fmt.Sprintf("Can't convert to string: %v", cv.Value()))
+	if s, ok := cv.TryAsString(); ok {
+		return s
 	}
-	return s
+
+	panic(fmt.Sprintf("Can't convert to string: %v", cv.Value()))
 }
 
 // AsInt returns the configuration value as an int, or panics if not
 // int-able
 func (cv Value) AsInt() int {
-	s, ok := cv.TryAsInt()
-	if !ok {
-		panic(fmt.Sprintf("Can't convert to int: %T %v", cv.Value(), cv.Value()))
+	if s, ok := cv.TryAsInt(); ok {
+		return s
 	}
-	return s
+
+	panic(fmt.Sprintf("Can't convert to int: %T %v", cv.Value(), cv.Value()))
```

Comment:
Yeah, I was just fixing everything that was off in this file, and will move it to a second PR.

Unfortunately the headache will remain, ~250 lines need to be added and removed, because we didn't make it right in the first time and kept adding bandaids over it.